network/dial/dial_task.go

```diff
@@ -0,0 +1,18 @@
+package dial
+
+import "github.com/libp2p/go-libp2p-core/peer"
+
+type DialTask struct {
+	index int
+
+	// info of the task
+	addr *peer.AddrInfo
+
+	// priority of the task (the higher the better)
+	priority uint64
+}
+
+// GetTaskInfo returns the peer information associated with the dial
+func (dt *DialTask) GetTaskInfo() *peer.AddrInfo {
```

Comment:
Why TaskInfo instead of Addr or AddrInfo?

Reply:
Agreed 💯 

Fixed:

[ad08c14](https://github.com/0xPolygon/polygon-edge/pull/343/commits/ad08c145fc150e4f2deffcb33e7147042b6f58c0)